version: '3.8'

services:
  postgres:
    image: postgres:alpine3.19
    restart: always
    environment:
      - POSTGRES_DB=${PGDATABASE:?error}
      - POSTGRES_USER=${PGUSER:?error}
      - POSTGRES_PASSWORD=${PGPASSWORD:?error}
    volumes:
      - db_data:/var/lib/postgresql/data
      - ./seed/products:/docker-entrypoint-initdb.d
    ports:
      - "${PGPORT}:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  db_data:
